### DISCUSSION TITLE
How to check if a variable is not defined, inside an expression, without getting Error: Undefined symbol

### DISCUSSION THREAD

**Discussion by tedeschia** [josdejong/mathjs#2970] [Q&A] [answered]
Is it possible to check for undefined variables inside an expression, before the expression evaluation throws the error "Undefined symbol"?

I'm using the lib to run expressions written by the user, and injecting some external variables as scope.

But I would need to let the user check if some variable exists in the expression logic, and proceed accordingly.

ie:
someVar==undefined?someDefaultResult:someVar+1.

If I try this kind of expression, it doesn't get evaluated, throwing the "Undefined symbol" error instead

**Comment by josdejong** [comment, marked as answer]
See: #2865, #1414, #660

Maybe `onUndefinedSymbol` is all you need.

**Reply by tedeschia** [reply]
yes, it was a very simple solution:
`mathInstance.SymbolNode.onUndefinedSymbol = () => null;`
thanks!
